src/PowerShellEditorServices/Services/DebugAdapter/Handlers/SemanticTokens.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.PowerShell.EditorServices.Services;
+using Microsoft.PowerShell.EditorServices.Services.TextDocument;
+using System.Management.Automation.Language;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document.Proposals;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models.Proposals;
+
+namespace Microsoft.PowerShell.EditorServices.Handlers
+{
+#pragma warning disable 618
+    internal class SemanticTokens : SemanticTokensHandler
+    {
+        private readonly ILogger _logger;
+        public WorkspaceService _workspaceService;
+        public SemanticTokens(ILogger<SemanticTokens> logger, WorkspaceService workspaceService) : base(new SemanticTokensRegistrationOptions() {
+            DocumentSelector = DocumentSelector.ForLanguage("powershell"),
+            Legend = new SemanticTokensLegend(),
+            DocumentProvider = new Supports<SemanticTokensDocumentProviderOptions>(true,
+                new SemanticTokensDocumentProviderOptions() {
+                    Edits = true
+                }),
+            RangeProvider = true
+        })
```

Comment:
The `SemanticTokensRegistrationOptions` object should probably be a `static readonly` field that just gets passed in here